app/shop.py

```diff
@@ -0,0 +1,29 @@
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self.name = name
+        self. location = location
+        self.products = products
+
+    def products_cost(self, client_products: dict) -> int:
+        total = 0
+        for product_name, amount in client_products.items():
+            if product_name in self.products:
+                total += self.products[product_name] * amount
+            else:
+                return -1

```

Comment:
Returning -1 when a product is not available is a valid approach, but ensure that the calling code handles this case to avoid incorrect calculations or logic errors.